src/nodejs/hooks/seeker_agent_test.go

```diff
@@ -0,0 +1,281 @@
+package hooks_test
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cloudfoundry/libbuildpack"
+	"github.com/cloudfoundry/nodejs-buildpack/src/nodejs/hooks"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("seekerHook", func() {
+	var (
+		err      error
+		buildDir string
+		depsDir  string
+		depsIdx  string
+		logger   *libbuildpack.Logger
+		stager   *libbuildpack.Stager
+		buffer   *bytes.Buffer
+		seeker   hooks.SeekerAfterCompileHook
+	)
+
+	BeforeEach(func() {
+		buildDir, err = ioutil.TempDir("", "nodejs-buildpack.build.")
+		Expect(err).To(BeNil())
+
+		depsDir, err = ioutil.TempDir("", "nodejs-buildpack.deps.")
+		Expect(err).To(BeNil())
+
+		depsIdx = "07"
+		err = os.MkdirAll(filepath.Join(depsDir, depsIdx), 0755)
+
+		buffer = new(bytes.Buffer)
+		logger = libbuildpack.NewLogger(buffer)
+
+		command := MockSeekerCommand{}
+
+		seeker = hooks.SeekerAfterCompileHook{
+			Command:    command,
+			Log:        logger,
+			Downloader: getMockedAgentDownloader(),
+			Unzipper:   getAgentUnzipper(),
+		}
+	})
+
+	JustBeforeEach(func() {
+		args := []string{buildDir, "", depsDir, depsIdx}
+		stager = libbuildpack.NewStager(args, logger, &libbuildpack.Manifest{})
+	})
+
+	AfterEach(func() {
+
+		err = os.RemoveAll(buildDir)
+		Expect(err).To(BeNil())
+
+		err = os.RemoveAll(depsDir)
+		Expect(err).To(BeNil())
+		os.RemoveAll(filepath.Join(os.TempDir(), "seeker_tmp"))
+	})
+	Describe("AfterCompile - obtain agent by directly downloading the agent", func() {
+		var (
+			oldVcapApplication string
+			oldVcapServices    string
+			oldBpDebug         string
+		)
+		BeforeEach(func() {
+			oldVcapApplication = os.Getenv("VCAP_APPLICATION")
+			oldVcapServices = os.Getenv("VCAP_SERVICES")
+			oldBpDebug = os.Getenv("BP_DEBUG")
+		})
+		AfterEach(func() {
+
+			os.Setenv("VCAP_APPLICATION", oldVcapApplication)
+			os.Setenv("VCAP_SERVICES", oldVcapServices)
+			os.Setenv("BP_DEBUG", oldBpDebug)
+		})
+
+		Context("VCAP_SERVICES contains seeker service - as a user provided service", func() {
+			BeforeEach(func() {
+				os.Setenv("VCAP_APPLICATION", `{"name":"pcf app"}`)
+				os.Setenv("VCAP_SERVICES", `{
+																			 "user-provided": [
```

Comment:
This probably shouldn't be indented this much. Looks okay in my editor, but different folks have different settings for tab indentation and as you can probably see here, Github does not render this well.

Reply:
fixed indentation